app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -343,6 +343,16 @@ def save_lans_inventory(switch, hashes)
     child_keys = [:subnets]
 
     save_inventory_multi(switch.lans, hashes, :use_association, [:uid_ems], child_keys, extra_keys)
+    store_ids_for_new_records(switch.lans, hashes, :uid_ems)
+
+    hashes.each do |h|
+      next if h[:id].nil?
+
+      parent_id = h.fetch_path(:parent, :id)
+      next if parent_id.nil?
+
+      switch.lans.find(h[:id]).update_attributes(:parent_id => parent_id)
```

Comment:
@Ladas I'm sure you can think of a better way to do this, but I saved everything first then update the parent_id so there aren't any issues with order dependency and made the disconnects simpler.  WDYT?

Reply:
``` Lan.where(:id => h[:id]).update_all(:parent_id => parent_id) ```

the query before was doing extra join and extra select, then update. This will do the same in 1 update query.